platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -272,23 +272,33 @@ private ImportMigrationConfigurationAdminEntry proxy(ImportMigrationEntry entry)
       } finally {
         IOUtils.closeQuietly(is);
       }
-      try {
-        // note: we also remove factory pid and pid from the dictionary as we do not want to restore
-        // those later
-        final String pid = Objects.toString(properties.remove(Constants.SERVICE_PID), null);
+      // note: we also remove factory pid and pid from the dictionary as we do not want to restore
+      // those later
+      final String pid = Objects.toString(properties.remove(Constants.SERVICE_PID), null);
+
+      if (pid == null) {
+        // this should never happen unless someone created the zip file manually and messed up the
+        // config file
+        context
+            .getReport()
+            .record(
+                new MigrationException("Import error: missing pid from configuration [%s].", path));
```

Comment:
❓ Do we have a unit test to make sure a `MigrationException` is recorded if the service PID is `null`?

Reply:
The config admin migratable only unit test at a higher level by running import followed by export. Since this case would only ever happen if someone was to manually tweak the zip file, I am less concern. In any case, we are recording an error which will abort the import by design. Not too worried here.